# *Q4*
-  TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.

In [18]:
# Assume,
# Null Hypothesis as Ho: Ho: Customer order forms defective % does not varies by centre.
# Alternate Hypothesis as Ha: Customer order forms defective % varies by centre.

## *Chi-Square Test*
- Chi2 contengency independence test("Goodness of fit test"):is used to test for the independence of two variable.
- It makes a comparison between the Expected Frequencies and the the observed frequencies.

In [19]:
#Import Labraries
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency
from sklearn import preprocessing




In [20]:
#Import Dataset
OrderForm = pd.read_csv(r"Costomer+OrderForm (1).csv")
OrderForm

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [21]:
label_encoder = preprocessing.LabelEncoder()

In [22]:
OrderForm["Phillippines"] = label_encoder.fit_transform(OrderForm["Phillippines"])
OrderForm["Indonesia"] = label_encoder.fit_transform(OrderForm["Indonesia"])
OrderForm["Malta"] = label_encoder.fit_transform(OrderForm["Malta"])
OrderForm["India"] = label_encoder.fit_transform(OrderForm["India"])

In [23]:
OrderForm["Phillippines"]  # [1 : Error Free, 0: Defective]
        

0      1
1      1
2      1
3      1
4      1
      ..
295    1
296    1
297    1
298    1
299    1
Name: Phillippines, Length: 300, dtype: int32

In [24]:
OrderForm["Phillippines"].value_counts()

1    271
0     29
Name: Phillippines, dtype: int64

In [25]:
OrderForm["Indonesia"].value_counts()

1    267
0     33
Name: Indonesia, dtype: int64

In [26]:
OrderForm["Malta"].value_counts()

1    269
0     31
Name: Malta, dtype: int64

In [27]:
OrderForm["India"].value_counts()    # [1: Error Free, 0: Defective]

1    280
0     20
Name: India, dtype: int64

In [28]:
# Make a Contingency Table
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [29]:
chi2 = chi2_contingency(obs)  # Return (Chi2 stats value,pvalue,df,expected obs)
chi2

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [30]:
pvalue=chi2[1]
pvalue

0.2771020991233135


- alpha(Significance level) = 5% = 0.05
- pvalue = 0.2771
- Comparing Both the values->>>>     pvalue>alpha(5% significance level)

In [31]:
# Decision: Failed to reject Null Hypothesis(Ho: Customer order forms defective % does not varies by centre.)
print("Customer order forms defective % does not varies by centre.")

Customer order forms defective % does not varies by centre.
